src/test/regress/spec/isolation_shard_rebalancer_progress.spec

```diff
@@ -118,11 +98,38 @@ step "s4-shard-move-sep-block-writes"
 	SELECT citus_move_shard_placement(1500009, 'localhost', 57637, 'localhost', 57638, shard_transfer_mode:='block_writes');
 }
 
+// Running two shard moves at the same time can cause racy behaviour over who
+// gets the lock. For the test where this move is used in we don't rely on the
+// advisory locks. So we disable taking the advisory lock there to avoid the
+// racy lock acquisition with the other concurrent move.
+step "s4-shard-move-sep-block-writes-without-advisory-locks"
+{
+	BEGIN;
```

Comment:
Instead of this, can we change the lock mode such that two concurrent moves do not block each other but only a concurrent move is blocked by concurrent `ConflictOnlyWithIsolationTesting`?

For example, we use `ShareLock` on `ConflictOnlyWithIsolationTesting` and keep `pg_advisory_lock` (which uses exclusive lock)?